### ISSUE TITLE
Doesn't install on Python 3.4

### ISSUE THREAD

**Issue by ionelmc** [getpelican/pelican#1609]
Pip fails with decoding errors cause Pelican includes the test suite in the package. And the test suite has weird characters in the tests directories. Eg: https://github.com/getpelican/pelican/tree/master/pelican/tests/output/custom_locale/posts/2012/f%C3%A9vrier

Sample output:
```
ERROR: invocation failed (exit code 2), logfile: C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\log\gen-1.log
ERROR: actionid=gen
msg=getenv
cmdargs=[local('C:\\Users\\ionel_000\\Projects\\blog.ionelmc.ro\\.tox\\gen\\Scripts\\pip.EXE'), 'install', '--pre', 'pelican==3.5.0']
env={'TMP': 'C:\\Users\\IONEL_~1\\AppData\\Local\\Temp', 'COMPUTERNAME': 'JME-PC', 'VS110COMNTOOLS': 'C:\\Program Files (x86)\\Microsoft Visual Studio 11.0\\Common7\\Tools\\', 'USERDOMAIN': 'JME-PC', 'VS100COMNTOOLS': 'C
:\\Program Files (x86)\\Microsoft Visual Studio 10.0\\Common7\\Tools\\', 'PSMODULEPATH': 'C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules\\', 'COMMANDER_PATH': 'C:\\Totalcmd', 'COMMONPROGRAMFILES': 'C:\\Program F
iles\\Common Files', 'PROCESSOR_IDENTIFIER': 'Intel64 Family 6 Model 42 Stepping 7, GenuineIntel', 'PROGRAMFILES': 'C:\\Program Files', 'PROCESSOR_REVISION': '2a07', 'SYSTEMROOT': 'C:\\Windows', 'PATH': 'C:\\Users\\ionel
_000\\Projects\\blog.ionelmc.ro\\.tox\\gen\\Scripts;C:\\ProgramData\\Oracle\\Java\\javapath;C:\\Program Files (x86)\\NVIDIA Corporation\\PhysX\\Common;c:\\Programs\\;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System3
2\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;c:\\Python27\\Scripts\\;C:\\Program Files\\TortoiseSVN\\bin;c:\\Python27\\;C:\\Program Files\\nodejs;C:\\Program Files\\TortoiseHg\\;C:\\Program Files\\Google\\Clo
ud SDK\\google-cloud-sdk\\bin;C:\\Program Files (x86)\\Git\\cmd;C:\\Program Files\\Microsoft Windows Performance Toolkit\\;C:\\Users\\ionel_000\\AppData\\Roaming\\Python\\Scripts;C:\\Users\\ionel_000\\AppData\\Roaming\\n
pm;C:\\Program Files (x86)\\Google\\google_appengine\\;c:\\PyPy\\', 'PROGRAMFILES(X86)': 'C:\\Program Files (x86)', 'LANG': 'en_US', 'VIRTUAL_ENV': 'C:\\Users\\ionel_000\\Projects\\blog.ionelmc.ro\\.tox\\gen', 'TEMP': 'C
:\\Users\\IONEL_~1\\AppData\\Local\\Temp', 'COMMONPROGRAMFILES(X86)': 'C:\\Program Files (x86)\\Common Files', 'PROCESSOR_ARCHITECTURE': 'AMD64', 'ALLUSERSPROFILE': 'C:\\ProgramData', 'PYTHONHASHSEED': '848', 'LOCALAPPDA
TA': 'C:\\Users\\ionel_000\\AppData\\Local', 'HOMEPATH': '\\Users\\ionel_000', 'USERDOMAIN_ROAMINGPROFILE': 'JME-PC', 'VS120COMNTOOLS': 'C:\\Program Files (x86)\\Microsoft Visual Studio 12.0\\Common7\\Tools\\', 'PROGRAMW
6432': 'C:\\Program Files', 'USERNAME': 'ionel_000', 'LOGONSERVER': '\\\\MicrosoftAccount', 'PROMPT': 'C\x08L\x08I\x08N\x08K\x08 \x08$P$G', 'COMSPEC': 'C:\\Windows\\system32\\cmd.exe', 'PROGRAMDATA': 'C:\\ProgramData', '
PYTHONPATH': '.', 'COMMANDER_DRIVE': 'C:', 'COMMANDER_EXE': 'C:\\Totalcmd\\TOTALCMD64.EXE', 'SCITE_HOME': 'C:\\ProgramData\\SciTE', 'SESSIONNAME': 'Console', 'PATHEXT': '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.
MSC;.PY', 'FP_NO_HOST_CHECK': 'NO', 'WINDIR': 'C:\\Windows', 'OPENSSL_CONF': 'C:\\OpenSSL-Win64\\bin\\openssl.cfg', 'HOMEDRIVE': 'C:', 'APPDATA': 'C:\\Users\\ionel_000\\AppData\\Roaming', 'COMMANDER_INI': 'C:\\Users\\ion
el_000\\wincmd.ini', 'SYSTEMDRIVE': 'C:', 'NUMBER_OF_PROCESSORS': '8', 'PROCESSOR_LEVEL': '6', 'COMMONPROGRAMW6432': 'C:\\Program Files\\Common Files', 'OS': 'Windows_NT', 'PUBLIC': 'C:\\Users\\Public', 'USERPROFILE': 'C
:\\Users\\ionel_000'}
Collecting pelican==3.5.0
Collecting python-dateutil (from pelican==3.5.0)
  Downloading python_dateutil-2.4.0-py2.py3-none-any.whl (175kB)
Collecting docutils (from pelican==3.5.0)
Collecting pytz>=0a (from pelican==3.5.0)
Collecting unidecode (from pelican==3.5.0)
Collecting six>=1.4 (from pelican==3.5.0)
Collecting feedgenerator>=1.6 (from pelican==3.5.0)
Collecting blinker (from pelican==3.5.0)
Collecting jinja2>=2.7 (from pelican==3.5.0)
Collecting pygments (from pelican==3.5.0)
  Downloading Pygments-2.0.2-py3-none-any.whl (672kB)
Collecting markupsafe (from jinja2>=2.7->pelican==3.5.0)
Installing collected packages: markupsafe, pygments, jinja2, blinker, feedgenerator, six, unidecode, pytz, docutils, python-dateutil, pelican











  Exception:
  Traceback (most recent call last):
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\basecommand.py", line 232, in main
      status = self.run(options, args)
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\commands\install.py", line 347, in run
      root=options.root_path,
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\req\req_set.py", line 549, in install
      **kwargs
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\req\req_install.py", line 740, in install
      self.move_wheel_files(self.source_dir, root=root)
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\req\req_install.py", line 949, in move_wheel_files
      isolated=self.isolated,
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\site-packages\pip\wheel.py", line 390, in move_wheel_files
      for row in reader:
    File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\encodings\cp1250.py", line 23, in decode
      return codecs.charmap_decode(input,self.errors,decoding_table)[0]
  UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 7822: character maps to <undefined>


ERROR: could not install deps [pelican==3.5.0]; v = InvocationError('C:\\Users\\ionel_000\\Projects\\blog.ionelmc.ro\\.tox\\gen\\Scripts\\pip.EXE install --pre pelican==3.5.0 (see C:\\Users\\ionel_000\\Projects\\blog.ion
elmc.ro\\.tox\\gen\\log\\gen-1.log)', 2)
___________________________________ summary ___________________________________
ERROR:   gen: could not install deps [pelican==3.5.0]; v = InvocationError('C:\\Users\\ionel_000\\Projects\\blog.ionelmc.ro\\.tox\\gen\\Scripts\\pip.EXE install --pre pelican==3.5.0 (see C:\\Users\\ionel_000\\Projects\\b
log.ionelmc.ro\\.tox\\gen\\log\\gen-1.log)', 2)
```

**Comment by jkw** [comment]
Must be a Windows-related issue. Works fine for me in ArchLinux with Python 3.4.2.

**Comment by blakev** [comment]
This is a `pip` issue, and can be resolved by adding `nl['encoding'] = 'utf-8'` on line 65 in `pip\wheel.py`

**Comment by MinchinWeb** [comment]
Ran into this issue tonight trying to upgrade to Pelican 3.6.0 from 3.5.0 using pip from the Windows command line. The pertinent line in the above output is:

~~~
File "C:\Users\ionel_000\Projects\blog.ionelmc.ro\.tox\gen\lib\encodings\cp1250.py", line 23, in decode
      return codecs.charmap_decode(input,self.errors,decoding_table)[0]
  UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 7822: character maps to <undefined>
~~~

Broken down, Windows command line is using [code page 1252](https://en.wikipedia.org/wiki/Windows-1252), which contains a very limited subset of Unicode (remember, this was originally used for Windows 1.0). `pip` tries to feed it the character `0x81`, but this isn't defined by the code page, so it breaks.

This same issue existed when upgrading to Pelican 3.5.0.

[More reading](http://www.i18nqa.com/debug/bug-double-conversion.html)

**Comment by MinchinWeb** [comment]
This is also a duplicate of #1545

**Comment by justinmayer** [comment]
Addressed via #1778. Fix will be included in next release.
